### ISSUE TITLE
Phase 1: LAMB optimizer + batch128 + label smoothing + AVX2 SIMD (20-30% speedup)

### ISSUE THREAD

**Issue by gHashTag** [gHashTag/trinity#293]
## Goal
Accelerate HSLM training by 20-30% without changing infrastructure.

### 1. LAMB Optimizer (replace AdamW)
- Implement LAMB in `src/tri-api/` training loop
- Better convergence on large batch sizes
- Expected: 5-10% faster + stability

### 2. Batch Size 128 + LR Scaling
- Increase from current batch size to 128
- Linear LR scaling: `new_lr = base_lr * (new_batch / old_batch)`
- Expected: 10-15% faster

### 3. Label Smoothing (eps=0.1)
- Add label smoothing to cross-entropy loss
- Expected: 5-10% PPL improvement

### 4. AVX2 SIMD in Zig matmul
- Widen from 8-wide to 32-wide vectors in `src/vsa.zig`
- Use `@Vector(32, i8)` for ternary ops
- Expected: 2-3x matmul speedup

## Reference
- BitNet b1.58: RMSNorm before linear layers critical for ternary
- Current baseline: ~24h to PPL=125 on Railway CPU
- Target: PPL=125 in ~17-19h

## Priority
P0 - start immediately, 1-2 days effort

### Files to modify

- src/tri-api/ (training loop, optimizer)
- src/vsa.zig (SIMD vectors)
- src/vm.zig (matmul kernel)

### Verification steps

1. `zig build test` passes
2. Training run with LAMB shows lower loss than AdamW baseline at same step
3. Batch=128 runs without OOM on Railway 32GB
4. AVX2 matmul benchmark shows 2x+ improvement

**Comment by gHashTag** [comment]
🌅 **Trinity Agent** | 2026-03-12T05:55:59Z
📋 **Step**: 1/8 — Authenticating with GitHub
🔄 **Status**: AWAKENING
⏱️ **Elapsed**: 0s

**Comment by gHashTag** [comment]
⚡ **Trinity Agent Dashboard** — Issue #293

| Field | Value |
|-------|-------|
| **Status** | CODING |
| **Step** | 4/8 |
| **Detail** | Claude Code running (timeout: 3600s) |
| **Elapsed** | 14s |
| **Container** | agent-293 |
| **Updated** | 2026-03-12T05:56:27Z |

**Comment by gHashTag** [comment]
☠️ **Trinity Agent Dashboard** — Issue #293

| Field | Value |
|-------|-------|
| **Status** | KILLED |
| **Step** | 2/8 |
| **Detail** | Container terminated by signal |
| **Elapsed** | 3s |
| **Container** | agent-293 |
| **Updated** | 2026-03-12T19:39:18Z |

**Comment by gHashTag** [comment]
⚡ **Trinity Agent** | 2026-03-12T19:39:35Z
📋 **Step**: 4/8 — Claude Code running (timeout: 3600s)
🔄 **Status**: CODING
⏱️ **Elapsed**: 13s

**Comment by gHashTag** [comment]
📖 **Trinity Agent** | 2026-03-12T19:53:08Z
📋 **Step**: 3/8 — Reading issue #293
🔄 **Status**: READING
⏱️ **Elapsed**: 7s

**Comment by gHashTag** [comment]
🌅 **Trinity Agent** | 2026-03-12T20:32:26Z
📋 **Step**: 1/8 — Authenticating with GitHub
🔄 **Status**: AWAKENING
⏱️ **Elapsed**: 0s

**Comment by gHashTag** [comment]
📖 **Trinity Agent Dashboard** — Issue #293

| Field | Value |
|-------|-------|
| **Status** | READING |
| **Step** | 3/8 |
| **Detail** | Reading issue #293 |
| **Elapsed** | 10s |
| **Container** | agent-293 |
| **Updated** | 2026-03-12T20:32:36Z |

**Comment by gHashTag** [comment]
❌ **Trinity Agent**: Compilation gate FAILED after 3 repair attempts. Code does not build.

```
timeout: failed to run command 'zig': No such file or directory
```

PR creation skipped. Issue needs manual attention.

**Comment by gHashTag** [comment]
🌅 **Trinity Agent Dashboard** — Issue #293

| Field | Value |
|-------|-------|
| **Status** | AWAKENING |
| **Step** | 2/8 |
| **Detail** | Creating worktree from bare repository |
| **Elapsed** | 3s |
| **Container** | agent-293 |
| **Updated** | 2026-03-12T21:15:53Z |

**Comment by gHashTag** [comment]
🌅 **Trinity Agent Dashboard** — Issue #293

| Field | Value |
|-------|-------|
| **Status** | AWAKENING |
| **Step** | 1/8 |
| **Detail** | Authenticating with GitHub |
| **Elapsed** | 0s |
| **Container** | agent-293 |
| **Updated** | 2026-03-12T21:52:53Z |

**Comment by gHashTag** [comment]
☠️ **Trinity Agent Dashboard** — Issue #293

| Field | Value |
|-------|-------|
| **Status** | KILLED |
| **Step** | 4/8 |
| **Detail** | Container terminated by signal |
| **Elapsed** | 14s |
| **Container** | agent-293 |
| **Updated** | 2026-03-12T22:00:26Z |

**Comment by gHashTag** [comment]
🌅 **Trinity Agent Dashboard** — Issue #293

| Field | Value |
|-------|-------|
| **Status** | AWAKENING |
| **Step** | 2/8 |
| **Detail** | Creating worktree from bare repository |
| **Elapsed** | 3s |
| **Container** | agent-293 |
| **Updated** | 2026-03-12T22:00:34Z |

**Comment by gHashTag** [comment]
❌ **Trinity Agent**: Compilation gate FAILED after 3 repair attempts. Code does not build.

```
timeout: failed to run command 'zig': No such file or directory
```

PR creation skipped. Issue needs manual attention.

**Comment by gHashTag** [comment]
🌅 **Trinity Agent Dashboard** — Issue #293

| Field | Value |
|-------|-------|
| **Status** | AWAKENING |
| **Step** | 1/8 |
| **Detail** | Authenticating with GitHub |
| **Elapsed** | 0s |
| **Container** | agent-293 |
| **Updated** | 2026-03-12T22:09:21Z |

**Comment by gHashTag** [comment]
📖 **Trinity Agent Dashboard** — Issue #293

| Field | Value |
|-------|-------|
| **Status** | READING |
| **Step** | 3/8 |
| **Detail** | Reading issue #293 |
| **Elapsed** | 8s |
| **Container** | agent-293 |
| **Updated** | 2026-03-12T22:09:29Z |

**Comment by gHashTag** [comment]
⚡ **Trinity Agent** | 2026-03-12T22:09:34Z
📋 **Step**: 4/8 — Claude Code running (timeout: 3600s)
🔄 **Status**: CODING
⏱️ **Elapsed**: 13s
